examples/04_New Models.ipynb

```diff
@@ -577,14 +577,115 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "## Direct Models"
+    "## Derived Parameters"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "## Derived Parameters"
+    "In the previous section, we constructed a new model from scratch, by subclassing from [progpy.PrognosticsModel](https://nasa.github.io/progpy/api_ref/prog_models/PrognosticModel.html#prog_models.PrognosticsModel) and specifying all of the necessary model components. An additional optional feature of PrognosticsModels is derived parameters, and we illustrate this below. \n",
+    "\n",
+    "A derived parameter is a parameter that is a function of another parameter. For example, in the case of a thrown object, one could assume that throwing speed is a function of thrower height, with taller throwing height resulting in faster throwing speeds. When such derived parameters exist, they must be updated whenever the parameters they depend on are updated. In PrognosticsModels, this is achieved with the `derived_params` method. "
```

Comment:
It might make sense to give some real examples from physical systems before we discuss our simple example. We can move the description of the example with ThrownObject to the next block (starting with "for this example")

For example in the electrochemistry battery model there are parameters for the maximum and minimum charge at the surface and bulk. This is dependent on the capacity of the battery (qMax). Therefore, we make these parameters derived from qMax and other independent parameters. 

It might also be worth mentioning that this feature can be used to cache combinations of parameters that are used frequently in state transition or other model methods. Created lumped parameters using derived parameters causes them to be calculated once when configuring, instead of each time step in simulation/prediction.

Reply:
Also, let's add a reference to the Parameters section in 01 Simulation